osu.Game/Skinning/SkinComponentsContainer.cs

```diff
@@ -8,32 +8,40 @@
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
-using osu.Game.Screens.Play.HUD;
 
 namespace osu.Game.Skinning
 {
-    public partial class SkinnableTargetContainer : SkinReloadableDrawable, ISkinnableTarget
+    /// <summary>
+    /// A container which holds many skinnable components, with functionality to add, remove and reload layouts.
+    /// Used to allow user customisation of skin layouts.
+    /// </summary>
+    /// <remarks>
+    /// This is currently used as a means of serialising skin layouts to files.
+    /// Currently, one json file in a skin will represent one <see cref="SkinComponentsContainer"/>, containing
+    /// the output of <see cref="ISerialisableDrawableContainer.CreateSerialisedInfo"/>.
+    /// </remarks>
+    public partial class SkinComponentsContainer : SkinReloadableDrawable, ISerialisableDrawableContainer
     {
         private Container? content;
 
-        public GlobalSkinComponentLookup.LookupType Target { get; }
+        public SkinComponentsContainerLookup Lookup { get; }
 
-        public IBindableList<ISkinnableDrawable> Components => components;
+        public IBindableList<ISerialisableDrawable> Components => components;
 
-        private readonly BindableList<ISkinnableDrawable> components = new BindableList<ISkinnableDrawable>();
+        private readonly BindableList<ISerialisableDrawable> components = new BindableList<ISerialisableDrawable>();
 
         public override bool IsPresent => base.IsPresent || Scheduler.HasPendingTasks; // ensure that components are loaded even if the target container is hidden (ie. due to user toggle).
 
         public bool ComponentsLoaded { get; private set; }
 
         private CancellationTokenSource? cancellationSource;
 
-        public SkinnableTargetContainer(GlobalSkinComponentLookup.LookupType target)
+        public SkinComponentsContainer(SkinComponentsContainerLookup lookup)
```

Comment:
This lookup-target switcheroo is a bit confusing. Would it be possible to get xmldoc on `Lookup` here and/or on `SkinComponentsContainerLookup.Target`? It's a bit too abstract for me to fully grasp as is.

Reply:
I named the inner one `Target` only to avoid `something.Lookup.Lookup`, which seems weird. I'm open to suggestion if you have anything better.

Reply:
I can understand not wanting `.Lookup.Lookup` for sure, and I don't necessarily have anything better to suggest. I guess what I'd be looking for here at most is explicit xmldoc on a few members.

Something like so:

```diff
diff --git a/osu.Game/Skinning/SkinComponentsContainer.cs b/osu.Game/Skinning/SkinComponentsContainer.cs
index 2620b40729..7e7a3c8ae6 100644
--- a/osu.Game/Skinning/SkinComponentsContainer.cs
+++ b/osu.Game/Skinning/SkinComponentsContainer.cs
@@ -24,6 +24,9 @@ public partial class SkinComponentsContainer : SkinReloadableDrawable, ISerialis
     {
         private Container? content;
 
+        /// <summary>
+        /// The <see cref="SkinComponentsContainerLookup"/> which will be used to retrieve components from the active skin.
+        /// </summary>
         public SkinComponentsContainerLookup Lookup { get; }
 
         public IBindableList<ISerialisableDrawable> Components => components;
diff --git a/osu.Game/Skinning/SkinComponentsContainerLookup.cs b/osu.Game/Skinning/SkinComponentsContainerLookup.cs
index ac6e98b23a..55d36d2040 100644
--- a/osu.Game/Skinning/SkinComponentsContainerLookup.cs
+++ b/osu.Game/Skinning/SkinComponentsContainerLookup.cs
@@ -3,8 +3,14 @@
 
 namespace osu.Game.Skinning
 {
+    /// <summary>
+    /// Represents a lookup of a collection of elements that make up a particular skinnable <see cref="TargetArea"/> of the game.
+    /// </summary>
     public class SkinComponentsContainerLookup : ISkinComponentLookup
     {
+        /// <summary>
+        /// The skinnable <see cref="TargetArea"/> of the game for which skin components will be returned.
+        /// </summary>
         public readonly TargetArea Target;
 
         public SkinComponentsContainerLookup(TargetArea target)
@@ -12,6 +18,9 @@ public SkinComponentsContainerLookup(TargetArea target)
             Target = target;
         }
 
+        /// <summary>
+        /// Represents a particular area or part of a game screen whose layout can be customised using the skin editor.
+        /// </summary>
         public enum TargetArea
         {
             MainHUDComponents,
```

Reply:
I've applied this, although removed self references (I feel it's a bit weird, like defining a word in a dictionary using the word itself).